src/shared/components/selected-species/SpeciesLozenge.test.tsx

```diff
@@ -0,0 +1,370 @@
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { render } from '@testing-library/react';
+import upperFirst from 'lodash/upperFirst';
+
+import { humanGenome } from './fixtures/speciesTestData';
+
+import SpeciesLozenge from './SpeciesLozenge';
+
+describe('<SpeciesLozenge />', () => {
+  describe('common-name_assembly-name', () => {
+    it('Displays common name and assembly name if common name is available', () => {
+      const { container } = render(
+        <SpeciesLozenge
+          species={humanGenome}
+          theme="blue"
+          speciesNameDisplayOption="common-name_assembly-name"
+        />
+      );
+
+      const renderedText = container.textContent;
+      const expectedText = `${humanGenome.common_name}${humanGenome.assembly.name}`;
+      expect(renderedText).toBe(expectedText);
+    });
+
+    it('Displays scientific name and assembly name if common name is not available', () => {
+      const testGenome = {
+        ...humanGenome,
+        common_name: null
+      };
+
+      const { container } = render(
+        <SpeciesLozenge
+          species={testGenome}
+          theme="blue"
+          speciesNameDisplayOption="common-name_assembly-name"
+        />
+      );
+
+      const renderedText = container.textContent;
+      const expectedText = `${humanGenome.scientific_name}${humanGenome.assembly.name}`;
+      expect(renderedText).toBe(expectedText);
+    });
+  });
+
+  describe('common-name_type_assembly-name', () => {
+    it('Shows common name, type, reference and assembly name if all present', () => {
+      const { container } = render(
+        <SpeciesLozenge
+          species={humanGenome}
+          theme="blue"
+          speciesNameDisplayOption="common-name_type_assembly-name"
+        />
+      );
+
+      const renderedText = container.textContent;
+      const commonName = humanGenome.common_name;
+      const typeKind = upperFirst(humanGenome.type.kind);
+      const typeValue = humanGenome.type.value;
+      const ref = 'Reference';
+      const assName = humanGenome.assembly.name;
+      const expectedText = `${commonName}${typeKind} - ${typeValue}, ${ref}${assName}`;
+      expect(renderedText).toBe(expectedText);
+    });
+
+    it('Shows scientific name, type, reference and assembly name if common name is absent', () => {
+      const testGenome = {
+        ...humanGenome,
+        common_name: null
+      };
+
+      const { container } = render(
+        <SpeciesLozenge
+          species={testGenome}
+          theme="blue"
+          speciesNameDisplayOption="common-name_type_assembly-name"
+        />
+      );
+
+      const renderedText = container.textContent;
+      const scientificName = testGenome.scientific_name;
+      const typeKind = upperFirst(testGenome.type.kind);
+      const typeValue = testGenome.type.value;
+      const ref = 'Reference';
+      const assName = testGenome.assembly.name;
```

Comment:
Hope there was no pun intended here 😆  Can you change this variable name to `assemblyName`?

Reply:
✅